pkg/ctl/create/cluster.go

```diff
@@ -424,6 +425,24 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
 			logger.Info("cluster should be functional despite missing (or misconfigured) client binaries")
 		}
 
+		if len(cfg.IAMIdentityMappings) > 0 {
+			logger.Info("iamIdentityMappings configuration detected, updating the auth ConfigMap")
+
+			m, err := mappingactions.New(cfg, clientSet, ctl, cmd.ProviderConfig.Region)
+			if err != nil {
+				return err
+			}
+
+			for _, mapping := range cfg.IAMIdentityMappings {
+				if err := mapping.Validate(); err != nil {
+					return err
+				}
+				if err := m.Create(ctx, mapping); err != nil {
+					return err
+				}
+			}
+		}
+
```

Comment:
This should go in a task, created in `CreateExtraClusterConfigTasks` (which has examples of other post-cluster-creation tasks). This can be appended as a task after the `IdentityProviders` task.

Reply:
@cPu1, that function has the following comment `// CreateExtraClusterConfigTasks returns all tasks for updating cluster configuration not depending on the control plane availability`. This updates the `aws-auth`  configmap so it does depend on control plane availability.

Reply:
That comment is outdated. The first task created in `CreateExtraClusterConfigTasks` waits for the control plane availability, so subsequent tasks can issue requests to the API server. Please feel free to update that comment, or I can do it later.

Reply:
Let's update the comment while we are here :)

Reply:
@cPu1 and @Himangini , I have updated the comment and converted that code to be a task.